packages/block-library/src/navigation-menu/editor.scss

```diff
@@ -52,79 +52,87 @@
 			background: none;
 		}
 	}
-}
-
-.wp-block-navigation-menu .block-editor-block-list__layout,
-.wp-block-navigation-menu {
-	display: flex;
-	white-space: nowrap;
-}
 
-.wp-block-navigation-menu__inserter-content {
-	padding: $grid-size-large;
-}
+	.wp-block-navigation-menu .block-editor-block-list__layout,
+	.wp-block-navigation-menu {
+		display: flex;
+		white-space: nowrap;
+	}
 
-.wp-block-navigation-menu-item {
-	margin-left: $grid-size;
-	.wp-block-navigation-menu-item__link {
-		margin-left: $grid-size-large;
+	.wp-block-navigation-menu__inserter-content {
+		padding: $grid-size-large;
 	}
-}
 
-/**
- * Colors Selector component
- */
-$colors-selector-size: 22px;
-.block-library-colors-selector {
-	width: auto;
-
-	// Toolbar colors-selector button.
-	.block-library-colors-selector__toggle {
-		display: block;
-		margin: 0 auto;
-		padding: 3px;
-		width: auto;
+	// Handling item colors depending on the navigation.
+	.wp-block-navigation-menu {
+		&.has-background-color {
+			.wp-block-navigation-menu-item {
```

Comment:
These don't need to be nested

Reply:
Generally, I'd agree with this. However in this case I'd say the usage _is_ valid.

To be clear, I generally avoid `&.` unless I'm handling a "state" selector (eg: `is-selected` or `is-hovered`). Outside of this scenario I avoid because it's potentially confusing. 

In this case we've got x2 "state" selectors:

1. `has-background`
2. `has-text-color`

Using `&.` allows us to visually group the common selectors together and avoid noise without introducing too much confusion or complex Sass syntax.

It's subjective but on balance, I'd vote to keep it.